modules/ROOT/pages/advanced-mule-concepts.adoc

```diff
@@ -0,0 +1,17 @@
+= Advanced Mule Concepts
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+:keywords: advanced, mule, concepts, runtime, engine
+
+This section covers some advanced Mule topics that will be useful once you are familiar with the basics of Mule application development and Mule runtime engine concepts.
+
+Learn about the xref:execution-engine.adoc[Execution Engine] to understand how the Mule runtime engine processes the operations in a Mule application’s flow, and understand how to change the thread pools if needed. Mule runtime engine uses execution metadata provided by operations in a flow to perform auto-tuning.
+
+The xref:about-classloading-isolation.adoc[Classloading Isolation] mechanism implemented in Mule 4 is designed to make upgrades easier by preventing components from exposing their internal resources. All components interact with each other through their defined APIs, meaning that only exposed resources can be accessed by other components. Classloading isolation ensures that a component's internal content is protected and can be changed anytime without affecting other components.
```

Comment:
```suggestion
The xref:about-classloading-isolation.adoc[classloading isolation] mechanism implemented in Mule 4 is designed to make upgrades easier by preventing components from exposing their internal resources. All components interact with each other through their defined APIs so that only exposed resources can be accessed by other components. Classloading isolation ensures that a component's internal content is protected and can be changed any time, without affecting other components.
```